libr/include/r_regex.h

```diff
@@ -13,8 +13,8 @@ typedef struct r_regex_t {
 } RRegex;
 
 typedef struct r_regmatch_t {
-	off_t rm_so;		/* start of match */
-	off_t rm_eo;		/* end of match */
+	st64 rm_so;		/* start of match */
```

Comment:
why is this failing? the change shuoldnt introduce any regression, but i dont know which is the issue that is trying to solve

Reply:
Due to other included headers, in regexp.c `__USE_FILE_OFFSET64` is defined before including r_regex.h so off_t is resolved to 4 byte, in regexec.c `__USE_FILE_OFFSET64` is not defined so off_t is resolved to 8 byte. You can run the commands I mentioned above to see the preprocess result and all visible macros on your rpi.

Reply:
I see :) good finding. Thanks!